File: frontend/src/interfaces/Reaction.ts
// src/interfaces/Reaction.ts
import type { User } from "./User";

export type TargetType = "thread" | "comment" | string; // ขยายได้ภายหลัง
export type ReactionType = "like" | "love" | "haha" | string;

export interface Reaction {
  ID: number;
  target_type: TargetType;
  target_id: number;
  type: ReactionType;
  user_id: number;
  user?: User;
}

export interface CreateReactionRequest {
  target_type: TargetType;
  target_id: number;
  type: ReactionType;
  user_id: number;
}

export interface UpdateReactionRequest {
  ID: number;
  target_type?: TargetType;
  target_id?: number;
  type?: ReactionType;
}

File: backend/entity/reaction.go
package entity

import (
	"gorm.io/gorm"
)


type Reaction struct {
	gorm.Model
	TargetType string `json:"target_type"` // "thread" | "comment" | อื่น ๆ
	TargetID   uint   `json:"target_id"`
	Type       string `json:"type"`        // "like", "love", ...
	UserID     uint   `json:"user_id"`
	User       *User  `gorm:"foreignKey:UserID" json:"user"`
}